Gems/PhysX/Code/Include/PhysX/HeightFieldAsset.cpp

```diff
@@ -30,9 +30,11 @@ namespace PhysX
 
         void HeightFieldAsset::SetHeightField(physx::PxHeightField* heightField)
```

Comment:
question: Might be outside the scope of this PR but as this looks like we're changing ownership, passing a `unique_ptr` by value here might be the better option to make things clearer ownership is changing (being moved)

Reply:
It's actually shared ownership between configuration and the asset, so not quite changing the ownership. 

It may be possible to use `shared_ptr` with custom deletion callbacks (we never delete heightfield pointers, only `Release()` them back to PhysX)

Reply:
Ah I see interesting, that might be nice (or to add ownership we could pass a `shared_ptr` by value)